src/vs/workbench/contrib/files/browser/fileCommands.ts

```diff
@@ -676,16 +677,9 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 		]
 	},
 	handler: async (accessor, args?: { viewType: string }) => {
-		const editorService = accessor.get(IEditorService);
-
-		if (typeof args?.viewType === 'string') {
-			const editorGroupsService = accessor.get(IEditorGroupsService);
-
-			const group = editorGroupsService.activeGroup;
-			await editorService.openEditor({ options: { override: args.viewType, pinned: true } }, group);
-		} else {
-			await editorService.openEditor({ options: { pinned: true } }); // untitled are always pinned
-		}
+		// {{SQL CARBON EDIT}} Modify to open untitled query editor
+		const queryEditorService = accessor.get(IQueryEditorService);
```

Comment:
So we don't need to use args?

Reply:
The args is just the viewType (which is just the editor mode/language) - but with how we're using this we only want it to open a query editor anyways so shouldn't be paying attention to it.

I toyed around with the idea of creating a new command instead and then replacing the file menu action with that one - but the problem there is that we'd also want to go through all the places that this one was used and see if we wanted to update that as well. And then if they added more things that called this in the future we'd have to find that out and fix those which gets annoying. Instead I thought it better to always override the behavior that when VS Code wants to create a new untitled file it opens an editor instead since that should be our default typically.